### PR TITLE
GSYE-452: Integrate validation with AVRO schemas

### PR DESCRIPTION
**Pull request by mrsaemir**
## Proposed changes

**INTEGRATION_TESTS_BRANCH**=feature/GSYE-452
**GSY_FRAMEWORK_BRANCH**=bug/GSYE-452

### CODE REVIEW

File: src/gsy_e/gsy_e_core/sim_results/endpoint_buffer.py [hunk 1]
```diff
@@ -45,6 +45,9 @@
     "max": None
 }
 
+scm_simulation_raw_data_validator = get_schema_validator("scm_simulation_raw_data_v1")
+simulation_raw_data_validator = get_schema_validator("simulation_raw_data_v1")
```
Comment (hannesdiedrich) [inline src/gsy_e/gsy_e_core/sim_results/endpoint_buffer.py:49]: If [this discussion](https://github.com/gridsingularity/gsy-framework/pull/421#discussion_r1086319896) reveals that we keep this interface: How about moving these into the constructor of the respective class? This way the SCM validator is not initialised when we have a normal simulation.

**Review by hannesdiedrich — COMMENTED** [review]
One comment, LGTM in general

File: src/gsy_e/gsy_e_core/sim_results/endpoint_buffer.py (hunk 1 above)
Comment (mrsaemir) [reply src/gsy_e/gsy_e_core/sim_results/endpoint_buffer.py:49]: Regardless of the result of our other discussion, it's possible that I move this into the constructor of the EndpointBuffer class if you agree.

Comment (mrsaemir) [reply src/gsy_e/gsy_e_core/sim_results/endpoint_buffer.py]: Done.

File: src/gsy_e/gsy_e_core/sim_results/endpoint_buffer.py [hunk 2]
```diff
@@ -45,8 +45,22 @@
     "max": None
 }
 
-scm_simulation_raw_data_validator = get_schema_validator("scm_simulation_raw_data_v1")
-simulation_raw_data_validator = get_schema_validator("simulation_raw_data_v1")
+
+class SimulationResultValidator:
```
Comment (hannesdiedrich) [inline src/gsy_e/gsy_e_core/sim_results/endpoint_buffer.py:49]: I do not understand the need for a separate class actually. Why not adding 
```python
self.results_validator = get_schema_validator("scm_simulation_raw_data")
```
into `CoefficientEndpointBuffer.__init__()`
and 
```python
self.results_validator = get_schema_validator("simulation_raw_data")
```
into `SimulationEndpointBuffer.__init__()`
?

Comment (mrsaemir) [reply src/gsy_e/gsy_e_core/sim_results/endpoint_buffer.py:49]: I did this for a couple of reasons:
1- There are more schema validators to be added in the future (for state variables, bids_offers_trades and configuration_tree). I felt that putting them all in the endpoint buffer will make the constructor of these classes even bigger.
2- I wanted to hide validation details from the endpoint buffers. This class does some sort of abstraction here.
3- (Personal pereference) I felt better to put all the real validation logic in one place to be able to manage them better in the future.

Comment (hannesdiedrich) [reply src/gsy_e/gsy_e_core/sim_results/endpoint_buffer.py:49]: Ok, after all this is only minor. Thanks for your explanation!

**Review by hannesdiedrich — APPROVED** [review]
LGTM

**Pull request merged**